apps/schema/src/components/ColumnEditModal.vue

```diff
@@ -463,9 +463,12 @@ export default {
       this.loading = true;
       if (this.column.refSchema) {
         this.client = Client.newClient(this.column.refSchema, this.$axios);
-        const schema = await this.client.fetchSchemaMetaData((error) => {
-          this.error = error;
-        });
+        const useCache = false;
```

Comment:
Ah, interesting. Does this mean that when I say 'newClient' I actually get a stale client? That is not what we want right? Each client instance should have its own cache!!!. If that is not possible then we should probably change the Client implementation?

Anyway, in principle we here don't need to do useCache is false because this call is used to retrieve in case of reference other schemas while editing current schema. Unless we are scared that users have multiple schema editors open and will mainwhile save these changes elsewhere we don't need to worry about cache.

But as I say above, I do worry if the cache is magically shared between client instances. However, to prevent that each client instance then needs to retrieve schema I would propose instead that we make a default client instance a property in the Vue app so all components within one app can use it if they so desire.

Reply:
i will remove the useCache is false here , as discussed this is meant as application level cache , not a per client level cache ( the client has the schema name as state , the app used the client.ts module to keep the application cache state ( note this means if used in ssr context different users share this state, a good thing in this context , but see notes on how to use things like vuex and pina in ssr context )